### PR TITLE
Code style improvements and lint fixes.

### PR DESCRIPTION
**Pull request by a-moreira**
This PR is mainly a code style suggestion, with some other small fixes.

- Proposes a `.rustfmt.toml` with  standardized default formatting based on the one we selected [for Floresta](https://github.com/Davidson-Souza/Floresta/blob/master/rust-toolchain.toml), which is small subset of the formatting settings used by rust-bitcoin projects.
- Some small fixes to make Clippy happy :-)
- Remove lock file, since this project is a library, and not a binary, as suggested by this [Stack Overflow question](https://stackoverflow.com/questions/62861623/should-cargo-lock-be-committed-when-the-crate-is-both-a-rust-library-and-an-exec) and the [Cargo FAQ](https://doc.rust-lang.org/cargo/faq.html#why-have-cargolock-in-version-control).

### CODE REVIEW

**Comment by a-moreira** [comment]
just fixed some small details, it's good to go @kcalvinalvin @Davidson-Souza

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -27,17 +27,26 @@ jobs:
   linting:
 
     runs-on: ubuntu-latest
-
+
```
Comment (Davidson-Souza) [inline .github/workflows/rust.yml:30]: nit: leading whitespace.

Comment (a-moreira) [reply .github/workflows/rust.yml]: done

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -53,4 +61,3 @@ jobs:
           toolchain: ${{ matrix.rust }}
 
       - name: Cross compile
-        run: cargo test --verbose --features ${{ matrix.feature }}
```
Comment (Davidson-Souza) [inline .github/workflows/rust.yml:56]: Question: Why did you remove the cross compilation tests?

Comment (a-moreira) [reply .github/workflows/rust.yml]: hmm my bad, I didn't see that. Will bring it back :-)

Comment (a-moreira) [reply .github/workflows/rust.yml]: @Davidson-Souza done!

**Comment by Davidson-Souza** [comment]
ACK [3e76bb5](https://github.com/mit-dci/rustreexo/pull/43/commits/3e76bb52a2face2528f864edcd7490d0b7481ed8)

**Pull request merged**